replica/database.cc

```diff
@@ -1792,6 +1792,7 @@ future<mutation> database::do_apply_counter_update(column_family& cf, const froz
     tracing::trace(trace_state, "Applying counter update");
     co_await apply_with_commitlog(cf, m, timeout);
 
+    co_await utils::get_local_injector().inject("apply_counter_update_delay_5s", std::chrono::seconds(5));
```

Comment:
I noticed that in release mode this translates to co_await make_ready_future<>(), which isn't a no-op (it adds a preemption check). Please post a separate pull request to fix this.

Reply:
@avikivity , how do you imagine the fix should look like? Should we change the return type of `utils::get_local_injector().inject()` to something that does not preempt when awaited? There are some places in the code that expect this to return a `future<>` (i.e. they call `.then()` or `.get()` on the result).

Reply:
You can use #ifndef the execute this code only in non-release builds.
Or, if there's a synchronous inline function that checks if fault injection is enabled and returms constant false in modes that do not support it, the compiler should optimize it away as unreachable code.

Reply:
> Or, if there's a synchronous inline function that checks if fault injection is enabled and returms constant false in modes that do not support it, the compiler should optimize it away as unreachable code.

Yes, there is `.enter()` which does exactly that.

Reply:
I changed it to use `.enter()`, thanks